internal/domain/review_suggestion.go

```diff
@@ -0,0 +1,27 @@
+package domain
+
+import (
+	"github.com/invopop/jsonschema"
+)
+
+type ReviewSuggestion struct {
+	Comment  string `json:"comment"`
+	Position int    `json:"position"`
+}
+
+type ReviewSuggestionsResponse struct {
+	ReviewSuggestions []ReviewSuggestion `json:"review_suggestions"`
+}
+
+func GenerateSchema[T any]() interface{} {
+	reflector := jsonschema.Reflector{
+		AllowAdditionalProperties: false,
```

Comment:
The function `GenerateSchema` uses reflection, which can introduce performance overhead. If the types being reflected are not expected to change frequently, consider caching the generated schema instead of generating it every time the function is called. This could lead to performance improvements, especially in cases where this function is invoked multiple times.